opal/datatype/opal_convertor.c

```diff
@@ -470,7 +470,8 @@ int32_t opal_convertor_set_position_nocheck( opal_convertor_t* convertor,
             }                                                             \
         }                                                                 \
         convertor->remote_size *= convertor->count;                       \
-        convertor->use_desc = &(datatype->desc);                          \
```

Comment:
At this point the convertor use_desc points to the optimized description, which is a description without datatype information (aka. not suitable for heterogeneous operations). I do think that forcing the switch to the default description (which contains the datatype information) is the right thing here. However, I would remove this line from the OPAL_CONVERTOR_COMPUTE_REMOTE_SIZE macro and instead put it right in the ```if( ((convertor->flags &...``` in the OPAL_CONVERTOR_PREPARE macro.